xarray/convert.py

```diff
@@ -53,3 +65,153 @@ def set_cdms2_attrs(var, attrs):
     cdms2_var = cdms2.createVariable(var.values, axes=axes, id=dataarray.name)
     set_cdms2_attrs(cdms2_var, var.attrs)
     return cdms2_var
+
+
+def _pick_attrs(attrs, keys):
+    """ Return attrs with keys in keys list
+    """
+    return dict((k, v) for k, v in attrs.items() if k in keys)
+
+
+def _get_iris_args(attrs):
+    """ Converts the xarray attrs into args that can be passed into Iris
+    """
+    # iris.unit is deprecated in Iris v1.9
+    import cf_units
+    args = {'attributes': _filter_attrs(attrs, iris_forbidden_keys)}
+    args.update(_pick_attrs(attrs, ('standard_name', 'long_name',)))
+    unit_args = _pick_attrs(attrs, ('calendar',))
+    if 'units' in attrs:
+        args['units'] = cf_units.Unit(attrs['units'], **unit_args)
+    return args
+
+
+# TODO: Add converting bounds from xarray to Iris and back
+def to_iris(dataarray):
+    """ Convert a DataArray into a Iris Cube
+    """
+    # Iris not a hard dependency
+    import iris
+    from iris.fileformats.netcdf import parse_cell_methods
+    try:
+        from dask.array import ma as dask_ma
+        from dask.array import Array
+    except ImportError:
+        dask_ma = None
+
+    dim_coords = []
+    aux_coords = []
+
+    for coord_name in dataarray.coords:
+        coord = encode(dataarray.coords[coord_name])
+        coord_args = _get_iris_args(coord.attrs)
+        coord_args['var_name'] = coord_name
+        axis = None
+        if coord.dims:
+            axis = dataarray.get_axis_num(coord.dims)
+        if coord_name in dataarray.dims:
+            iris_coord = iris.coords.DimCoord(coord.values, **coord_args)
+            dim_coords.append((iris_coord, axis))
+        else:
+            iris_coord = iris.coords.AuxCoord(coord.values, **coord_args)
+            aux_coords.append((iris_coord, axis))
+
+    args = _get_iris_args(dataarray.attrs)
+    args['var_name'] = dataarray.name
+    args['dim_coords_and_dims'] = dim_coords
+    args['aux_coords_and_dims'] = aux_coords
+    if 'cell_methods' in dataarray.attrs:
+        args['cell_methods'] = parse_cell_methods(dataarray.attrs['cell_methods'])
+
+    # Create the right type of masked array (should be easier after #1769)
+    if isinstance(dataarray.data, Array):
+        masked_data = dask_ma.masked_invalid(dataarray)
```

Comment:
just import `from dask.array import ma as dask_ma` right about this line to avoid the need for guarding the import statement in try/except.

Reply:
Much nicer - thanks!